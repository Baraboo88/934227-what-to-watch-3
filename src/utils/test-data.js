export const promoFilmData = {
  promoFilmName: `The Grand Budapest Hotel`,
  promoFilmGenre: `Drama`,
  promoFilmDate: 2014
};

export const titleClickHandler = () => {};
