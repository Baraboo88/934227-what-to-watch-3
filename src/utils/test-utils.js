export const promoFilmData = {
  promoFilmName: `The Grand Budapest Hotel`,
  promoFilmGenre: `Drama`,
  promoFilmDate: 2014
};

export const titleClickHandler = () => {};

export const mockFilms = [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `./img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    img: `./img/bohemian-rhapsody.jpg`
  },
  {
    id: 3,
    name: `Macbeth`,
    img: `./img/macbeth.jpg`
  },
  {
    id: 4,
    name: `Aviator`,
    img: `./img/aviator.jpg`
  },
  {
    id: 5,
    name: `We need to talk about Kevin`,
    img: `./img/we-need-to-talk-about-kevin.jpg`
  }
];

export const findByAttr = (app, attr) => {
  return app.find(`[data-test='${attr}']`);
};
