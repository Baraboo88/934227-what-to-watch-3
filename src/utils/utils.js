import PropTypes from 'prop-types';

export const cardPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  avgMark: PropTypes.number.isRequired,
  descriptions: PropTypes.arrayOf(PropTypes.string.isRequired),
  director: PropTypes.string.isRequired,
  stares: PropTypes.arrayOf(PropTypes.string.isRequired),
  genre: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired
}).isRequired;
