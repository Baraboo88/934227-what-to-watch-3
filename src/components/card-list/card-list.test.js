import renderer from 'react-test-renderer';
import React from 'react';
import CardsList from "./cards-list";
import {mockFilms} from "../../utils/test-utils";

it(`CardList component is correctly rendering`, () => {
  const tree = renderer.create(<CardsList
    cards={mockFilms}
    onMouseOver = {() => {}}
    onTitleClick={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
