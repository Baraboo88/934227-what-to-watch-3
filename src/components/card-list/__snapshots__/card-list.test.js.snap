// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component is correctly rendering 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    data-test="small-card-test"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="./img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-test="test-title-click"
        href="#"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    data-test="small-card-test"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="./img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-test="test-title-click"
        href="#"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    data-test="small-card-test"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="./img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-test="test-title-click"
        href="#"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    data-test="small-card-test"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="./img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-test="test-title-click"
        href="#"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    data-test="small-card-test"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="./img/we-need-to-talk-about-kevin.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-test="test-title-click"
        href="#"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>,
]
`;
